src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -198,6 +206,11 @@ await this.LifeCycleNotificationHelper.OrchestratorStartingAsync(
             OrchestratorExecutionResult orchestratorResult;
             if (functionResult.Succeeded)
             {
+                if (workerRequiresHistory)
+                {
+                    throw new SessionAbortedException("The worker has since ended the extended session and needs an orchestration history to execute the orchestration request.");

```

Comment:
Does the In Process implementation of extended sessions also throw a `SessionAbortedException` in this scenario?

Reply:
It's not an issue in in-proc since there's no worker so there's no situation where a worker could evict an extended session before the host has (it's much easier for in-proc in that sense)